runcommandonset/src/utils.rs

```diff
@@ -0,0 +1,178 @@
+                exit(EXIT_INVALID_INPUT);
+            }
+        }
+    } else if let Some(executable) = executable {
+        command = runcommand::RunCommand {
+            executable,
+            arguments,
+            exit_code,
+        };
+    }
+    else {
+        error!("Error: Executable is required when input is not provided via stdin");
+        exit(EXIT_INVALID_INPUT);
+    }
+    command
+}
+
+/// Setup tracing subscriber based on the provided trace level and format.
+///
+/// # Arguments
+///
+/// * `trace_level` - The level of information of to output
+/// * `trace_format` - The format of the output
+///
+/// # Errors
+///
+/// If unable to initialize the tracing subscriber, an error message is printed and tracing is disabled.
+pub fn enable_tracing(trace_level: &TraceLevel, trace_format: &TraceFormat) {
+    // originally implemented in dsc/src/util.rs
+    let tracing_level = match trace_level {
+        TraceLevel::Error => Level::ERROR,
+        TraceLevel::Warning => Level::WARN,
+        TraceLevel::Info => Level::INFO,
+        TraceLevel::Debug => Level::DEBUG,
+        TraceLevel::Trace => Level::TRACE,
+    };
+
+    let filter = EnvFilter::try_from_default_env()
+        .or_else(|_| EnvFilter::try_new("warning"))
+        .unwrap_or_default()
+        .add_directive(tracing_level.into());
+    let layer = tracing_subscriber::fmt::Layer::default().with_writer(std::io::stderr);
+    let fmt = match trace_format {
+        TraceFormat::Default => {
+            layer
+                .with_ansi(true)
+                .with_level(true)
+                .with_line_number(true)
+                .boxed()
+        },
+        TraceFormat::Plaintext => {
+            layer
+                .with_ansi(false)
+                .with_level(true)
+                .with_line_number(false)
+                .boxed()
+        },
+        TraceFormat::Json => {
+            layer
+                .with_ansi(false)
+                .with_level(true)
+                .with_line_number(true)
+                .json()
+                .boxed()
+        }
+    };
+
+    let subscriber = tracing_subscriber::Registry::default().with(fmt).with(filter);
+
+    if tracing::subscriber::set_global_default(subscriber).is_err() {
+        eprintln!("Unable to set global default tracing subscriber.  Tracing is diabled.");
+    }
+}
+
+/// Invoke a command and return the exit code, stdout, and stderr.
+///
+/// # Arguments
+///
+/// * `executable` - The command to execute
+/// * `args` - Optional arguments to pass to the command
+///
+/// # Errors
+///
+/// Error message then exit if the command fails to execute or stdin/stdout/stderr cannot be opened.
+pub fn invoke_command(executable: &str, args: Option<Vec<String>>) -> (i32, String, String) {
+    // originally implemented in dsc_lib/src/dscresources/command_resource.rs
+    debug!("Invoking command {} with args {:?}", executable, args);
```

Comment:
This should probably be a trace as this is user supplied args rather than a resource being executed so may contain sensitive data